packages/x-date-pickers/src/models/fields.ts

```diff
@@ -174,6 +174,17 @@ export type BaseSingleInputFieldProps<
 > = BaseFieldProps<TValue, TDate, TSection, TEnableAccessibleFieldDOMStructure, TError> &
   BaseForwardedSingleInputFieldProps<TEnableAccessibleFieldDOMStructure>;
 
+export type BaseSingleInputFieldPropsInPickerSlots<
```

Comment:
This makes sure the `slots.field` can't pass `onOpen` that would break everything, we should probably remove other props from the interface at some point.